btb/tuning/hyperparams/base.py

```diff
@@ -27,38 +104,78 @@ def inverse_transform(self, values):
                 Single value or 2D ArrayLike of normalized values.
 
         Returns:
-            denormalized (Union[object, List[object]]):
-                Denormalized value or list of denormalized values.
+            denormalized (numpy.ndarray):
+                2D array of denormalized values.
         """
-        pass
+        values = to_array(values, self.K)
+        self._within_search_space(values)
+
+        return self._inverse_transform(values)
 
     @abstractmethod
     def sample(self, n_samples):
-        """Generate an array of ``num_samples`` random samples in the search space.
+        """Generate an array of ``n_samples`` random samples in the search space.
 
         Args:
             n_samples (int):
                 Number of values to sample.
 
         Returns:
-            samples (ArrayLike):
+            samples (numpy.ndarray):
                 2D array with of shape (n_samples, self.K)
         """
-        pass
 
-    @abstractmethod
     def transform(self, values):
         """Transform one or more hyperparameter values.
 
         Transform one or more hyperparameter values from the original hyperparameter space to the
         normalized search space [0, 1]^k.
+        The accepted value formats are:
+
+            - Single value:
+                A single value from the original hyperparameter space.
+            - List:
+                A list composed by values from the original hyperparameter space.
+            - 2D Array:
+                Two dimension array like object that contains values from the original
+                hyperparameter space.
 
         Args:
             values (Union[object, List[object]]):
                 Single value or list of values to normalize.
 
         Returns:
-            normalized (ArrayLike):
+            normalized (numpy.ndarray):
                 2D array of shape(len(values), self.K)
+
+        Examples:
+            >>> from btb.tuning.hyperparams.numerical import IntHyperParam
+            >>> ihp = IntHyperParam(min=1, max=4)
+            >>> ihp.transform(1)
+            array([[0.125]])
+
```

Comment:
Remove the empty lines. Also make sure to copy/paste from a python interpreter to make sure that it matches the actual python output.